Sources/Remote Logging/Crash Logging/CrashLogging.swift

```diff
@@ -225,64 +186,40 @@ public extension CrashLogging {
-        let semaphore = DispatchSemaphore(value: 0)
-
-        var networkError: Error?
-
-        try logErrorImmediately(error, userInfo: userInfo, level: level) { result in
-
-            switch result {
-                case .success:
-                    TracksLogDebug("💥 Successfully transmitted crash data")
-                case .failure(let err):
-                    networkError = err
-            }
-
-            semaphore.signal()
-        }
-
-        semaphore.wait()
-
-        if let networkError = networkError {
-            throw networkError
-        }
+        logErrorsImmediately([error], userInfo: userInfo, level: level, andWait: true, callback: {})
+        TracksLogDebug("💥 Events flush completed. When using Sentry, this either means all events were sent or that the flush timeout was reached.")
     }
 
-    /// Returns an array of threads for the current stack trace.  This hack is needed because we don't have
-    /// any public mechanism to access the stack trace threads to add them to our custom events.
-    ///
-    /// Ref: https://github.com/getsentry/sentry-cocoa/issues/1451#issuecomment-1240782406
-    ///
-    private func currentThreads() -> [Sentry.Thread] {
-        guard let client = SentrySDK.currentClient() else {
-            return []
+    private func logErrorsImmediately(
+        _ errors: [Error],
+        userInfo: [String: Any]? = nil,
+        level: SentryLevel = .error,
+        andWait wait: Bool,
+        callback: @escaping () -> Void
+    ) {
+        errors.forEach { error in
+            // Amending the global scope on a per-event basis seems like the best way to add the
+            // caller-provided `userInfo` and `level`.
+            SentrySDK.capture(error: error) { scope in
+                // Under the hood, `setExtras` uses `NSMutableDictionary` `addEntriesFromDictionary`
+                // meaning this won't replace the whole extras dictionary.
+                scope.setExtras(userInfo)
+                scope.setLevel(level)
+            }
         }
 
-        return client.currentThreads()
-    }
-}
-
-extension SentryDsn {
-    func getAuthString() -> String? {
-
-        guard let user = url.user else {
-            return nil
+        let flushEventThenCallCallback: () -> Void = {
+            SentrySDK.flush(timeout: CrashLogging.eventsFlushingTimeout)
+            callback()
         }
-
-        var data = [
-            "sentry_version=7",
-            "sentry_client=tracks-manual-upload/\(TracksLibraryVersion)",
-            "sentry_timesetamp=\(Date().timeIntervalSince1970)",
-            "sentry_key=\(user)",
-        ]
-
-        if let password = url.password {
-            data.append("sentry_secret=\(password)")
+        if wait {
+            flushEventThenCallCallback()
+        } else {
+            let queue = DispatchQueue(label: "com.automattic.tracks.flushSentry", qos: .background)
```

Comment:
Any reason why using a custom queue instead of `DispatchQueue.global`? Also, any reason why choose `.background` QoS, instead of the default one? (I almost always use the default QoS since I've heard scary things about QoS but don't really understand it 😸 )

Reply:
I followed the suggestion from one of the Sentry engineers from [here](https://github.com/getsentry/sentry-cocoa/issues/1013#issuecomment-1244958158).

I trust your advice on this matter, though. I'll update.

Reply:
Done in [Automattic/Automattic-Tracks-iOS@`8a0c84b` (#232)](https://github.com/Automattic/Automattic-Tracks-iOS/pull/232/commits/8a0c84b201c7b919aa6a2b8b752e0dcb3269c0a9)